app/models/user.rb

```diff
@@ -1,11 +1,16 @@
 class User < ApplicationRecord
-  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
+  before_save :downcase_email
 
-  validates :name, :password, :confirm_password, presence: true
-  validates :name, length: {minimum: Settings.length.min_10}
-  validates :password, length: {minimum: Settings.length.min_10}
+  validates :name, presence: true, length: {minimum: Settings.length.min_10}
   validates :email, presence: true, uniqueness: true,
-    length: {maximum: Settings.length.max_255},
-    format: {with: VALID_EMAIL_REGEX}
+  length: {maximum: Settings.length.max_255},
+  format: {with: Settings.regex.email}
+  validates :password, length: {minimum: Settings.length.min_10}
   has_secure_password
+
+  private
+
+  def downcase_email
+    self.email = email.downcase
```

Comment:
dùng `email.downcase!` được ko em?

Reply:
- email.downcase!: return new value and reassign to email variable nên sẽ dùng được (chỗ này viết tiếng việt hơi khó hiểu nên em viết như vậy ạ)